Post is failing

The following Github Action snippet is failing with the error below:

```yaml
    - name: Validate wheel cache
      uses: techservicesillinois/cache-validation@v1
      id: cache
      with:
        path: cache
        cache_hit: ${{ steps.ghcache.outputs.cache-hit }}
        remove_invalid_files: true
        touch: true
```

Github Action [error](https://github.com/techservicesillinois/awscli-login/runs/7796341076?check_suite_focus=true):

```
Post job cleanup.
EEXIST: file already exists, open 'cache/MD5SUMS'
Total files hashed 0.
```

For context the `Validate wheel cache` step returned:

```
Run techservicesillinois/cache-validation@v1
Cache miss: nothing to validate.
```

Unit tests fail with same error message on Node v16 using npm v8:

```
$ make
npm ci --also=dev
npm WARN config also Please use --include=dev instead.
npm WARN old lockfile 
npm WARN old lockfile The package-lock.json file was created with an old version of npm,
npm WARN old lockfile so supplemental metadata must be fetched from the registry.
npm WARN old lockfile 
npm WARN old lockfile This is a one-time fix-up, please be patient...
npm WARN old lockfile 
npm WARN deprecated source-map-url@0.4.1: See https://github.com/lydell/source-map-url#deprecated
npm WARN deprecated request-promise-native@1.0.9: request-promise-native has been deprecated because it extends the now deprecated request package, see https://github.com/request/request/issues/3142
npm WARN deprecated urix@0.1.0: Please see https://github.com/lydell/urix#deprecated
npm WARN deprecated har-validator@5.1.5: this library is no longer supported
npm WARN deprecated resolve-url@0.2.1: https://github.com/lydell/resolve-url#deprecated
npm WARN deprecated source-map-resolve@0.5.3: See https://github.com/lydell/source-map-resolve#deprecated
npm WARN deprecated sane@4.1.0: some dependency vulnerabilities fixed, support for node < 10 dropped, and newer ECMAScript syntax/features added
npm WARN deprecated uuid@3.4.0: Please upgrade  to version 7 or higher.  Older versions may use Math.random() in certain circumstances, which is known to be problematic.  See https://v8.dev/blog/math-random for details.
npm WARN deprecated request@2.88.2: request has been deprecated, see https://github.com/request/request/issues/3142

added 522 packages, and audited 523 packages in 6s

24 packages are looking for funding
  run `npm fund` for details

8 vulnerabilities (3 moderate, 2 high, 3 critical)

To address all issues, run:
  npm audit fix

Run `npm audit` for details.
npm run test

> cache-validation@1.0.0 test
> jest --collect-coverage

Browserslist: caniuse-lite is outdated. Please run:
npx browserslist@latest --update-db

Why you should do it regularly:
https://github.com/browserslist/browserslist#browsers-data-updating
(node:94878) [DEP0147] DeprecationWarning: In future versions of Node.js, fs.rmdir(path, { recursive: true }) will be removed. Use fs.rm(path, { recursive: true }) instead
(Use `node --trace-deprecation ...` to show where the warning was created)
 FAIL  ./index.test.js
  ● fail & remove file w/ remove_invalid_files set to true

    expect(received).toBe(expected) // Object.is equality

    Expected: false
    Received: true

       95 |
       96 |   // If remove_invalid_files is true then file foo should not exist
    >  97 |   expect(fs.existsSync(foo)).toBe(!object.remove_invalid_files &&
          |                              ^
       98 |                                   !object.remove_invalid_paths);
       99 |
      100 |   // File bar should never exist

      at testBadMissingFile (index.test.js:97:30)
      at Object.<anonymous> (index.test.js:120:3)

  ● fail on nonexistent MD5SUM & return 1 w/fatal set to true

    ENOENT: no such file or directory, open 'good/MD5SUMS'



  ● fail on nonexistent MD5SUM & return 0 w/fatal set to false

    ENOENT: no such file or directory, open 'good/MD5SUMS'



  ● Do not check MD5SUMS file if INPUT_CACHE_HIT is not true

    ENOENT: no such file or directory, open 'bad/MD5SUMS'



 PASS  ./md5sum.test.js
--------------|---------|----------|---------|---------|-------------------
File          | % Stmts | % Branch | % Funcs | % Lines | Uncovered Line #s 
--------------|---------|----------|---------|---------|-------------------
All files     |   98.84 |    97.67 |     100 |   98.84 |                   
 index.js     |   98.39 |     96.3 |     100 |   98.39 | 28                
 md5sum.js    |   96.43 |      100 |     100 |   96.43 | 58                
 testUtils.js |     100 |      100 |     100 |     100 |                   
 utils.js     |     100 |      100 |     100 |     100 |                   
--------------|---------|----------|---------|---------|-------------------

Test Suites: 1 failed, 1 passed, 2 total
Tests:       4 failed, 10 passed, 14 total
Snapshots:   0 total
Time:        1.696 s
Ran all test suites.
make: *** [Makefile:8: test] Error 1
TECH-P10R38812:cache-validation ddriddle$ 
TECH-P10R38812:cache-validation ddriddle$ npm --version
8.11.0
TECH-P10R38812:cache-validation ddriddle$ node --version
v16.16.0
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -4,6 +4,9 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+## [v1.1.2]
+- Fix failure on cache miss and existing MD5SUM ([#20])
+
 ## [v1.1.1]
 * Upgrade to Node v14 ([#19])
 
@@ -15,12 +18,14 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 ## [v1.0.0]
 * Initial release ([#1])
 
+[#20]: https://github.com/techservicesillinois/cache-validation/pull/20
 [#19]: https://github.com/techservicesillinois/cache-validation/pull/19
 [#14]: https://github.com/techservicesillinois/cache-validation/pull/14
 [#12]: https://github.com/techservicesillinois/cache-validation/pull/12
 [#11]: https://github.com/techservicesillinois/cache-validation/pull/11
 [#1]: https://github.com/techservicesillinois/cache-validation/pull/1
 
+[v1.1.2]: https://github.com/techservicesillinois/cache-validation/releases/tag/v1.1.2
 [v1.1.1]: https://github.com/techservicesillinois/cache-validation/releases/tag/v1.1.1
 [v1.1.0]: https://github.com/techservicesillinois/cache-validation/releases/tag/v1.1.0
 [v1.0.0]: https://github.com/techservicesillinois/cache-validation/releases/tag/v1.0.0

--- a/md5sum.js
+++ b/md5sum.js
@@ -13,7 +13,7 @@ const path = require('path');
 const util = require('./utils.js');
 
 function createMD5SUMS(directory) {
-  const file = fs.openSync(path.join(directory, "MD5SUMS"), 'wx');
+  const file = fs.openSync(path.join(directory, "MD5SUMS"), 'w');
 
   let count = 0;
   util.walkPath(directory, function(filePath, isDir) {

--- a/md5sum.test.js
+++ b/md5sum.test.js
@@ -59,17 +59,16 @@ test('generate empty MD5SUMS file in an empty directory', () => {
   expect(fs.readdirSync(tmpdir.name).length).toBe(1)
 });
 
-test('fail if MD5SUM exists', () => {
+// Regression #16: Overwrite existing MD5SUMS on cache miss
+test('Generate MD5SUMS even if MD5SUMS exists', () => {
   process.env['INPUT_CACHE_HIT'] = 'false';
   process.env['INPUT_PATH'] = 'good';
 
-  util.setupGoodDirectory({MD5SUMS: true});
-  md5sum.main();
+  const [hashes] = util.setupGoodDirectory({MD5SUMS: false});
+  fs.writeFileSync(path.join('good', 'MD5SUMS'), '');
 
-  expect(process.exitCode).toBe(1);
-  expect(console.log.mock.calls[0][0])
-    .toBe("EEXIST: file already exists, open " +
-          `'${path.join("good", "MD5SUMS")}'`);
+  md5sum.main();
+  expectOutput(hashes);
 });
 
 test('fail on nonexistent path', () => {

